src/Framework/TestCase.php

```diff
@@ -29,35 +29,31 @@ class TestCase
 	 * Runs the test case.
 	 * @return void
 	 */
-	public function run($method = NULL)
+	public function run()
 	{
-		$r = new \ReflectionObject($this);
+		if (func_num_args()) {
+			throw new \LogicException('Calling TestCase::run($method) is deprecated. Use TestCase::runTest($method) instead.');
+		}
+
 		$methods = array_values(preg_grep(self::METHOD_PATTERN, array_map(function (\ReflectionMethod $rm) {
 			return $rm->getName();
-		}, $r->getMethods())));
-
-		if (substr($method, 0, 2) === '--') { // back compatibility
-			$method = NULL;
-		}
+		}, (new \ReflectionObject($this))->getMethods())));
 
-		if ($method === NULL && isset($_SERVER['argv']) && ($tmp = preg_filter('#(--method=)?([\w-]+)$#Ai', '$2', $_SERVER['argv']))) {
+		# TODO: Support 'php test.phpt testMethod' or 'php test.phpt --method=testMethod' only?
+		if (isset($_SERVER['argv']) && ($tmp = preg_filter('#(--method=)?([\w-]+)$#Ai', '$2', $_SERVER['argv']))) {
```

Comment:
I am not using TestCase, but I think that it is not needed to have two ways how do the same think and it blocks first parameter.